common/src/test/java/org/conscrypt/TrustManagerImplTest.java

```diff
@@ -135,76 +219,33 @@ public void testHttpsEndpointIdentification() throws Exception {
             params.setEndpointIdentificationAlgorithm("HTTPS");
 
-            try {
-                tmi.getTrustedChainForServer(chain, "RSA",
+            try { // this should fail
+                certs = tmi.getTrustedChainForServer(chain, "RSA",
                     new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
+                 assertEquals(Arrays.asList(chain), certs);
                 fail();
-            } catch (CertificateException expected) {
-            }
-
-            certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(goodHostname, chain), params));
-            assertEquals(Arrays.asList(chain), certs);
-
-            // Override the global default hostname verifier with a Conscrypt-specific one that
-            // always passes.  Both scenarios should pass.
-            Conscrypt.setDefaultHostnameVerifier(new ConscryptHostnameVerifier() {
-                @Override public boolean verify(String s, SSLSession sslSession) { return true; }
-            });
-
-            certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
-            assertEquals(Arrays.asList(chain), certs);
-
-            certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(goodHostname, chain), params));
-            assertEquals(Arrays.asList(chain), certs);
 
-            // Now set an instance-specific verifier on the trust manager.  The bad hostname should
-            // fail again.
-            Conscrypt.setHostnameVerifier(tmi, new TestHostnameVerifier());
-
-            try {
-                tmi.getTrustedChainForServer(chain, "RSA",
-                    new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
-                fail();
             } catch (CertificateException expected) {
             }
-
-            certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(goodHostname, chain), params));
-            assertEquals(Arrays.asList(chain), certs);
-
-            // Remove the instance-specific verifier, and both should pass again.
-            Conscrypt.setHostnameVerifier(tmi, null);
-
-            certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(badHostname, chain), params));
-            assertEquals(Arrays.asList(chain), certs);
-
             certs = tmi.getTrustedChainForServer(chain, "RSA",
-                new FakeSSLSocket(new FakeSSLSession(goodHostname, chain), params));
+                        new FakeSSLSocket(new FakeSSLSession(goodHostname, chain), params));
             assertEquals(Arrays.asList(chain), certs);
-        } finally {
-            Conscrypt.setDefaultHostnameVerifier(null);
-            HttpsURLConnection.setDefaultHostnameVerifier(oldDefault);
+        } catch (Exception e) {
```

Comment:
You still need a `finally` clause with `Conscrypt.setDefaultHostnameVerifier(null);` or this test can break any tests which run after it

Reply:
thank you I added that back!